app/models/user.rb

```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+
+  has_many :coupons, dependent: :destroy_async
```

Comment:
why async?

Reply:
It will destroy all coupons by enqueuing an asynchronous job, so that we don't have to wait for coupons to be deleted

Reply:
Changed to destroy